vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/NumberField.java

```diff
@@ -234,4 +240,30 @@ public String getPattern() {
         return getPatternString();
     }
 
+    private static class Formatter implements SerializableFunction<Double, String> {
+
+        private final DecimalFormat decimalFormat = new DecimalFormat("#.#",
+            DecimalFormatSymbols.getInstance(Locale.ENGLISH));
```

Comment:
Can you explain why this second argument is needed? Is it to make sure that period is used as the decimal-separator regardless of the JVM locale?

Anyway, a comment in the code could be added because the question might rise up again.

Reply:
Yes, to avoid using the JVM Locale. I'll add a comment.

Reply:
Done

Reply:
I was concerned what will happen if the browser locale defines comma as the decimal separator. But it seems that even when the user can input the value with comma, the `value`-property is always with period, so everything works fine.

IMO it could be further clarified in the comment, e.g. add to the end: "The value-property always uses period as the decimal-separator regardless of the browser locale." or something along those lines.

Reply:
Thanks! :)